pallets/pips/src/lib.rs

```diff
@@ -809,194 +864,264 @@ decl_module! {
-        #[weight = (750_000_000, DispatchClass::Operational, Pays::Yes)]
-        pub fn override_referendum_enactment_period(origin, id: PipId, until: Option<T::BlockNumber>) -> DispatchResult {
-            let sender = ensure_signed(origin)?;
-            let current_did = Context::current_identity_or::<Identity<T>>(&sender)?;
-
-            // 1. Only release coordinator
-            ensure!(
-                Some(current_did) == T::GovernanceCommittee::release_coordinator(),
-                Error::<T>::BadOrigin);
-
-            Self::is_referendum_state(id, ReferendumState::Scheduled)?;
+        /// * `BadOrigin` - unless a GC voting majority executes this function.
+        /// * `CannotSkipPip` - a given PIP has already been skipped too many times.
+        /// * `SnapshotResultTooLarge` - on len(results) > len(snapshot_queue).
+        /// * `SnapshotIdMismatch` - if:
+        ///   ```
+        ///    ∃ (i ∈ 0..SnapshotQueue.len()).
+        ///      results[i].0 ≠ SnapshotQueue[SnapshotQueue.len() - i].id
+        ///   ```
+        ///    This is protects against clearing queue while GC is voting.
+        #[weight = (1_000_000_000, DispatchClass::Operational, Pays::Yes)]
+        pub fn enact_snapshot_results(origin, results: Vec<(PipId, SnapshotResult)>) -> DispatchResult {
+            T::VotingMajorityOrigin::ensure_origin(origin)?;
 
-            // 2. New value should be valid block number.
-            let next_block = <system::Module<T>>::block_number() + 1.into();
-            let new_until = until.unwrap_or(next_block);
-            ensure!( new_until >= next_block, Error::<T>::InvalidFutureBlockNumber);
+            let max_pip_skip_count = Self::max_pip_skip_count();
+
+            <SnapshotQueue<T>>::try_mutate(|queue| {
+                let mut to_bump_skipped = Vec::new();
+                let mut to_reject = Vec::new();
+                let mut to_approve = Vec::new();
+
+                // Go over each result...
+                for (id, action) in results.iter().copied() {
+                    match queue.pop() { // ...and "zip" with the queue in reverse.
+                        // An action is missing a corresponding PIP in the queue, bail!
+                        None => return Err(Error::<T>::SnapshotResultTooLarge.into()),
+                        // The id at queue position vs. results mismatches.
+                        Some(p) if p.id != id => return Err(Error::<T>::SnapshotIdMismatch.into()),
+                        // All is right...
+                        Some(_) => {},
+                    }
+                    match action {
+                        // Make sure the PIP can be skipped and enqueue bumping of skip.
+                        SnapshotResult::Skip => {
+                            let count = PipSkipCount::get(id);
+                            ensure!(count < max_pip_skip_count, Error::<T>::CannotSkipPip);
+                            to_bump_skipped.push((id, count + 1));
```

Comment:
We need increment `count` before looping the next result otherwise multiple skip results will only apply a total plus 1 to the count.

Reply:
This is defined per `PipId`; it won't affect other PIPs, and if you do `[(1, Skip), (1, Skip)]` you'll get an error on second element.